ovos_utils/json_helper.py

```diff
@@ -1,9 +1,74 @@
 import json
+from copy import copy
 from json_database.utils import is_jsonifiable, get_key_recursively, \
     get_key_recursively_fuzzy, get_value_recursively_fuzzy, \
     get_value_recursively, jsonify_recursively
 
 
+def nested_get(base, items):
+    """Access a nested object in base by item sequence."""
+    if not len(items):
+        return None
+    val = copy(base)
+    for key in items:
+        if key not in val:
+            return None
+        val = val[key]
```

Comment:
I'm a little confused here. `items`, I take it, is a list of keys which are expected to be present in the 'outer' dictionary (`base`).

The routine takes a deep copy of the 'outer' collection, then iterates over the 'inner' collection, checking whether *each* key in the 'inner' collection is present in (the deep copy of) the 'outer' collection.

If any key in the 'inner' collection is *missing* from the deep copy, it quits and returns `None`. However, if *all* the keys from the 'inner' collection are present in the deep copy, it will return the deep copy's value for the *final* key.

Is this the intended behavior?

Reply:
yes, exactly, the unittests should cover the intended usage i hope...

Reply:
To look at the unit test, it doesn't look like some of the tests should pass (which they do.)

```python
dct = {'a': 1, 'b': {'c': 1, 'd': 2}}
self.assertEqual(nested_get(dct, ["b", "c"]), 1)
```

```python
items = ["b", "c"]
for key in items:
  if key not in dct:
    return None
"c" in dct == False # so why doesn't this thing return None
```

Reply:
Ah, but for the override. These variable names are murderous 😛

Reply:
Might be more intuitive as something like `nested_get(parent_dict, target_element)`

Reply:
Perhaps even more intuitive as:

`nested_get(parent_dict, outer_element, inner_element)`, creating
`nested_get(dct, 'b', 'c') -> dict['b']['c']`

Reply:
the point is that items can have any length, so it really needs a list

Reply:
get key1 then get key2.... then get keyN

Reply:
could we just use `*args` instead of a list? I tend to agree that would be more intuitive (though a docstring for the variable works just as well to me)

Reply:
i also considered using args, but i find a list more intuitive and easier to pass along, i expect the input to these methods will always come from somewhere up the chain, so a list makes it easier to manage than args i think

Reply:
In favor of args, this really only becomes sugar when you don't know ahead of time what elements you're going to look for. Otherwise indices are roughly equivalent.

This being the case - and I can imagine plenty of situations where it *is* the case - I think we'll *frequently* unpack arguments into this thing. You're already assembling a list, sure, but the unpack lets you pull it from places other than a list, without necessarily casting *to* a list.

```
localized_signature = nested_get(localized_functions, *(module, lang_code or get_default_lang(), full_code or get_default_locale(), func_name))
```

tasty.